app/controllers/api/coins_controller.rb

```diff
@@ -1,18 +1,26 @@
 class Api::CoinsController < ApiController
 
   def index
-    query = params[:q] || {}
-    if params[:exclude_watched]
-      query[:id_not_in] = current_user.watchlist.coin_ids
+    if (!has_news_features?)
```

Comment:
@anthotsang Could we use `return respond_unfound if !has_news_features?` to keep the rest of the code untouched for easier diff?

Reply:
If that (or some variation thereof) works, please go ahead and implement across all invocations as well, thanks!